src/resources/views/base/inc/alerts.blade.php

```diff
@@ -1,21 +1,40 @@
 {{-- Bootstrap Notifications using Prologue Alerts & PNotify JS --}}
 <script type="text/javascript">
-  Noty.overrideDefaults({
-    layout   : 'topRight',
-    theme    : 'backstrap',
-    timeout  : 2500, 
-    closeWith: ['click', 'button'],
-  });
+    // This is intentionaly run after dom loads so this way we can avoid showing duplicate alerts
+    // when the user is beeing redirected by persistent table, that happens before this event triggers.
+    window.addEventListener('DOMContentLoaded', function() {
+        Noty.overrideDefaults({
+            layout: 'topRight',
+            theme: 'backstrap',
+            timeout: 2500,
+            closeWith: ['click', 'button'],
+        });
 
-  @foreach (\Alert::getMessages() as $type => $messages)
+        // get alerts from the alert bag
+        var $alerts_from_php = JSON.parse('@json(\Alert::getMessages())');
 
-      @foreach ($messages as $message)
+        // get the alerts from the localstorage
+        var $alerts_from_localstorage = JSON.parse(localStorage.getItem('backpack_alerts')) ?? {};
 
-        new Noty({
-          type: "{{ $type }}",
-          text: "{!! str_replace('"', "'", $message) !!}"
-        }).show();
+        // merge both php alerts and localstorage alerts
+        Object.entries($alerts_from_php).forEach(([type, messages]) => {
+            if(typeof $alerts_from_localstorage[type] !== 'undefined') {
+                $alerts_from_localstorage[type].push(...messages);
+            } else {
+                $alerts_from_localstorage[type] = messages;
+            }
+        });
 
-      @endforeach
-  @endforeach
-</script>
\ No newline at end of file
+        for (var type in $alerts_from_localstorage) {
+            for(var message in $alerts_from_localstorage[type]) {
+                new Noty({
+                    type: type,
+                    text: $alerts_from_localstorage[type][message]
+                }).show();
+            }
```

Comment:
```suggestion
            let messages = new Set($alerts_from_localstorage[type]);
            messages.forEach(text => new Noty({type, text}).show());
```
By using a `Set` we easily clean up repeated messages.

Reply:
This will clear our duplicate messages. Nice One!